mbuild/conversion.py

```diff
@@ -0,0 +1,1364 @@
+    Files are read using the MDTraj package unless the `use_parmed` argument is
+    specified as True. Please refer to http://mdtraj.org/1.8.0/load_functions.html
+    for formats supported by MDTraj and https://parmed.github.io/ParmEd/html/
+    readwrite.html for formats supported by ParmEd.
+
+    Parameters
+    ----------
+    filename_or_object : str, mdtraj.Trajectory, parmed.Structure, mbuild.Compound,
+            pybel.Molecule
+        Name of the file or topology from which to load atom and bond information.
+    relative_to_module : str, optional, default=None
+        Instead of looking in the current working directory, look for the file
+        where this module is defined. This is typically used in Compound
+        classes that will be instantiated from a different directory
+        (such as the Compounds located in mbuild.lib).
+    compound : mb.Compound, optional, default=None
+        Existing compound to load atom and bond information into.
+        New structure will be added to the existing compound as a sub compound.
+    coords_only : bool, optional, default=False
+        Only load the coordinates into an existing compound.
+    rigid : bool, optional, default=False
+        Treat the compound as a rigid body
+    backend : str, optional, default=None
+        Backend used to load structure from file. If not specified, a default
+        backend (extension specific) will be used.
+    smiles: bool, optional, default=False
+        Use Open Babel to parse filename as a SMILES string
+        or file containing a SMILES string.
+    infer_hierarchy : bool, optional, default=True
+        If True, infer hierarchy from chains and residues
+    **kwargs : keyword arguments
+        Key word arguments passed to mdTraj for loading.
+
+    Returns
+    -------
+    compound : mb.Compound
+
+    """
+    # First check if we are loading from an object)
+    if not isinstance(filename_or_object, str):
+        return load_object(
+                        object=filename_or_object,
+                        compound=compound,
+                        coords_only=coords_only,
+                        rigid=rigid,
+                        infer_hierarchy=infer_hierarchy,
+                        **kwargs)
+    # Second check if we are loading SMILES strings
+    elif smiles:
+        return load_smiles(
+                        smiles_or_filename=filename_or_object,
+                        compound=compound,
+                        infer_hierarchy=infer_hierarchy)
+    # Last, if none of the above, load from file
+    else:
+        return load_file(
+                        filename=filename_or_object,
+                        relative_to_module=relative_to_module,
+                        compound=compound,
+                        coords_only=coords_only,
+                        rigid=rigid,
+                        backend=backend,
+                        infer_hierarchy=infer_hierarchy,
+                        **kwargs)
```

Comment:
Pure curiosity -- what is the reason for the order of object, smiles, file?

Reply:
This order just follows the order in the original code, but now this order can be changed pretty easily. Ideally, they should be arranged in the most common to the least common used, but I am not sure which order they are.